server/src/main/java/se/fortnox/reactivewizard/server/RwServer.java

```diff
@@ -155,6 +158,18 @@ public static void registerShutdownDependency(Runnable blockShutdownUntil) {
         RwServer.blockShutdownUntil = blockShutdownUntil;
```

Comment:
Perhaps it would be possible to change this into a list of Runnable so you can use it instead and avoid duplicated code? I wrote this code some years ago and the only reason for not having a list was hat I didn't need it at that time.

Reply:
That was my first intention as well, but then I realised that we're not applying the handlers universally at one single point in the shutdown procedure. If it was like that, a list of Runnable would be a natural response. Instead, we attach handlers to the hooks at different stages. So waiting for dependencies is one stage, shutting down the logging is a different stage. It happens at the very last step of the procedure. So the Runnable handlers should be identifiable in order to call them where they are needed.

Reply:
The approach has changed. We're not using Runnable for this purpose. We inject an interface instead.